tests/functional/aws-node-sdk/test/versioning/replicationBucket.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const async = require('async');
+
+const withV4 = require('../support/withV4');
+const BucketUtility = require('../../lib/utility/bucket-util');
+
+const bucketName = `versioning-bucket-${Date.now()}`;
+
+
+function checkError(err, code) {
+    assert.notEqual(err, null, 'Expected failure but got success');
+    assert.strictEqual(err.code, code);
```

Comment:
Small thing that can be ignored, but the second test will fail if the first one fails as well. As if it was equal null would not have a code attribute. So you could remove the first assert and still cover all the points. But sometimes the error message might be a bit harder to debug. As it would just say "null does not have an attribute code" instead of "code does not match: code".

Reply:
It's exactly why it is. Even though the second test will fail if the first one fails, when the test is run in isolation it is helpful.

Reply:
ok good to know